### ISSUE TITLE
Promises break in "real" versions of node

### ISSUE THREAD

**Issue by ericlifka** [austinkelleher/giphy-api#9]
Promise detection isn't done before various "Promise.reject" paths are hit, meaning in _real_ node versions (like 0.9.12, why would anyone use anything else? I mean honestly...) using most of the api functions results in reference errors. See the _obviously correct_ example:
```
var giphy = require('giphy-api')();

giphy.search();
```

gives

```
                return Promise.reject('Search phrase cannot be empty.');
                       ^
ReferenceError: Promise is not defined
    at Object.GiphyAPI.search (/Users/ericlifka/test-giphy/node_modules/giphy-api/index.js:62:24)
    at Object.<anonymous> (/Users/ericlifka/test-giphy/index.js:3:7)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Function.Module.runMain (module.js:497:10)
    at startup (node.js:119:16)
    at node.js:891:3
```

This is clearly a huge blocker to using this library.

**Comment by philidem** [comment]
@ericlifka makes a good point. You should always maintain support Node 0.9.x. Why would you introduce promises to this library and not use callbacks? I guess I'll search for an alternative.

**Comment by austinkelleher** [comment]
![goteem](https://media.giphy.com/media/CYU3D3bQnlLIk/giphy.gif)

**Comment by xdumaine** [comment]
Honestly excepted more from a 1.x version of a library.

**Comment by austinkelleher** [comment]
Fixed in caeeea0. Let me know if you have any more issues with real versions of Node. Bumped version to 1.1.14.
